backend/service/topology/cache.go

```diff
@@ -12,13 +12,16 @@ import (
 	"google.golang.org/protobuf/encoding/protojson"
 
 	topologyv1 "github.com/lyft/clutch/backend/api/topology/v1"
+	"github.com/lyft/clutch/backend/service"
 )
 
 const topologyCacheLockId = "topology:cache"
 
 // Performs leader election by acquiring a postgres advisory lock.
 // Once the lock is acquired topology caching is started.
-func (c *client) acquireTopologyCacheLock(ctx context.Context) {
+func (c *client) acquireTopologyCacheLock() {
+	ctx, ctxCancelFunc := context.WithCancel(context.Background())
```

Comment:
I need to find a better way of handling / canceling the context here, open to any suggestions.